engine/execution/state/state.go

```diff
@@ -379,6 +390,70 @@ func (s *state) PersistStateInteractions(ctx context.Context, blockID flow.Ident
 	return operation.RetryOnConflict(s.db.Update, operation.InsertExecutionStateInteractions(blockID, views))
 }
 
+func (s *state) PersistExecutionState(ctx context.Context, header *flow.Header, endState flow.StateCommitment, chunkDataPacks []*flow.ChunkDataPack, executionResult *flow.ExecutionResult, events []flow.Event, serviceEvents []flow.Event, results []flow.TransactionResult) error {
+
+	span, childCtx := s.tracer.StartSpanFromContext(ctx, trace.EXESaveExecutionResults)
+	defer span.Finish()
+
+	blockID := header.ID()
+
+	// Write Batch is BadgerDB feature designed for handling lots of writes
+	// in efficient and automatic manner, hence pushing all the updates we can
+	// as tightly as possible to let Badger manage it.
+	// Note, that it does not guarantee atomicity as transactions has size limit
+	// but it's the closes thing to atomicity we could have
+	batch := s.db.NewWriteBatch()
+
+	for _, chunkDataPack := range chunkDataPacks {
+		err := s.chunkDataPacks.BatchStore(chunkDataPack, batch)
+		if err != nil {
+			return fmt.Errorf("cannot store chunk data pack: %w", err)
+		}
+	}
+
+	err := s.commits.BatchStore(blockID, endState, batch)
+	if err != nil {
+		return fmt.Errorf("cannot store state commitment: %w", err)
+	}
+
+	err = s.results.BatchStore(executionResult, batch)
+	if err != nil {
+		return fmt.Errorf("cannot store state commitment: %w", err)
+	}
+
+	err = s.results.BatchIndex(blockID, executionResult.ID(), batch)
```

Comment:
```suggestion
	// it overwrites the index if exists already
	err = s.results.BatchIndex(blockID, executionResult.ID(), batch)
```

It overwrites, right? better to note it as comment

Reply:
Sure!